Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+
+## New features added in V1 task:
+
+## **- Support for fetching latest version for a given major or major.minor version**
+This feature was being asked by users who wanted to install the latest version as soon as it becomes available, without manually changing the version in the task. For information about user's asks, refer [here](https://github.com/Microsoft/azure-pipelines-tasks/issues/9171).
+
+This feature is implemented by supporting patterns in version input: `version`. This input now accepts .Net sdk/runtime versions in the following patterns
+- `Major.Minor.PatchVersion :` complete version as present in releases.json for the Major.Minor Version channel. Example: 2.2.104
+- `Major.Minor.x :` Latest version in the Major.Minor version channel release. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used to get all latest bug fixes automatically in a major.minor version. As back-compat is maintained in major.minor version, thus ensuring applications being built will not break.
+- `Major.x :` Latest version released in the Major version channel. It can also be the latest preview version if `includePreviewVersions` input is enabled/true. This can be used by tools or library developers who need to test with every latest version for compatibility.
+
+### How it works:
+The correct version needs to be identified in case user enters pattern based version such as 2.2.x. This is how the exact needed version and its information is extracted:
+
+- [releases-index.json](https://github.com/dotnet/core/blob/master/release-notes/releases-index.json) The file containing links to all .Net Core release channels is download and read into a JSON object `releasesIndex`. This is used to find all available release channels and links to their releases.json.
+
+- The `version` input is divided into three sections:
+  - Major
+  - Minor
+  - Patch
+
+- Based on Minor versions value, the following can happen
+  - Numeric : we get the corresponding channel's (Major.Minor) releases.json link from `releasesIndex`. Then the releases.json is downloaded and parsed into an object: `channelInformation`.
+    - NOTE: in case exact version is not found, the function logs error and task will fail.
+  - x : in this case, we find the latest channel with `support-phase` != `preview` (unless `includePreviewVersions` is true). The releases.json for the selected channel is then downloaded and parsed into json object `releaseInformation`.
+    - The version with heighest `release.sdk/runtime.version` is then returned. (preview version will only be selected if `includePreviewVersions` is true)
```

Comment:
how will you calculate highest version - based on symantic versioning or using latest-runtime/latest-sdk property?

Reply:
Currently based on symantic versioning